### DISCUSSION TITLE
Utilizing MQTT to send one (and only one) snapshot per ID

### DISCUSSION THREAD

**Discussion by danlo315** [blakeblackshear/frigate#15495] [Ask A Question]
**Use Case**
I would like to quickly identify an event that's happening via a snapshot, followed by the clip at the conclusion of the event.  I'm going this route because there are situations where events are happening for long duration, and I would like to know during the event, not after it ends.

**Approach**
I'm using Home Assistant automation with the MQTT topic of `frigate/events` as trigger, sending out Telegram messages to my own "home channel".  I've leaned on this post as guidance:
https://github.com/blakeblackshear/frigate/issues/6551
- I am ignoring type: new because `has_snapshot == false`.
- I have a concluding automation for `type: end` trigger that sends out the clip.  I'm currently seeing no issues sending this out, with the exception that sometimes the clip seems to cut off too quickly (it's a different question that I'll ask separately).

**Issue**
The challenge I have is on the update messages.  Ideally, I would like to only send one snapshot; multiple snapshots clutter up my Telegram channel.
- The post I referenced above uses scores to send out the optimum snapshot, which is better than continuously sending out updates.  An edge case of only one update *and* the new message has a higher score means I will miss sending out even one update notification.
- I would like to see if there is a way to identify the **first** update message?  It won't be the most optimal snapshot but I think that should be fine.  I was looking at `['before']['end_time'] == null`, but I've seen multiple subsequent update messages that have the same null value.

Suggestions are great appreciated!

**Comment by NickM-27** [comment]
I would suggest just sending a snapshot when `before.has_snapshot == False and after.has_snapshot == True`

**Reply by danlo315** [reply]
That's a great idea, trying it out now!

**Reply by danlo315** [reply]
Thanks for the tip, and this mostly works.  However,  I utilize zones, and there is a (not so) edge case where the new event is not in a zone, but the subsequent updates are in the zone.  In this case, the first update message that contains the zone has `before.has_snapshot == true`.  I've tuned it to the following triggers/conditions and am stress testing it now.

```
triggers:
  - trigger: mqtt
    topic: frigate/events
    payload: "on"
    value_template: >-
      {{ "on" if "p_front_porch" in value_json["after"]["current_zones"] else
      "off"}}
conditions:
  - condition: template
    value_template: "{{ trigger.payload_json[\"type\"] == \"update\" }}"
  - condition: or
    conditions:
      - condition: template
        value_template: >-
          {{ trigger.payload_json["before"]["has_snapshot"] == false and
          trigger.payload_json["after"]["has_snapshot"] == true }}
      - condition: template
        value_template: >-
          {{ "p_front_porch" not in
          trigger.payload_json["before"]["entered_zones"] and
          trigger.payload_json["after"]["has_snapshot"] == true }}
```
